### DISCUSSION TITLE
多个子工程通过环境变量或者 option 控制导入

### DISCUSSION THREAD

**Discussion by davidxiaozhi** [xmake-io/xmake#4009] [Q&A]
### 你在什么场景下需要该功能？

多个子工程通过环境变量或者 option 控制导入,例如 GPu 和 CPU 下导入的工程的数量是不一样的,GPU key 包含 CPU, 但是 CPU 模式仅仅只有 CPU

### 描述可能的解决方案

例如
// cpu 模式或 gpu 模式
includes("cpu/")
includes("cpu_test/")
//only GPU 模式
includes("gpu/")

### 描述你认为的候选方案

例如
// cpu 模式或 gpu 模式
includes("cpu/")
includes("cpu_test/")
//only GPU 模式
includes("gpu/")

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
? 不知道你要说什么

[a comment by an automated account is not shown]

**Comment by davidxiaozhi** [comment]
@waruqi  如何在 xmake.lua 当中依据一个 option 或者环境变量的配置动态includes()
相对于
if ("$(device)"=="gpu") then
    includes("cpu")
    includes("gpu")
elseif ("$(device)"=="gcu") then
    includes("cpu")
    includes("gcu")
else
    includes("cpu")
end

**Comment by waruqi** [comment]
```lua
if is_config("device", "gpu") then
    includes("cpu", "gpu")
end
```

**Comment by waruqi** [comment]
https://xmake.io/#/zh-cn/manual/conditions?id=is_config
